test/Parser/TestParserPatternMatching.hs

```diff
@@ -0,0 +1,173 @@
+testParseFullPatternMatching = describe "parsePatternMatchesIntLiteral" $ do
+    it "parses multiple Integer -> Integer match cases separated by | and otherwise" $ do
+        parse parsePatternMatches "" "0 -> 0 | otherwise -> 1" `shouldBe` 
+                        Right (FullPatternMatch 
+                                [ PatternMatchLit (PLiteral (IntLiteral 0)) (IntLiteral 0) ]
+                                (OtherwiseLit (IntLiteral 1)))
+
+    it "parses multiple Integer -> Integer match cases separated by | and otherwise" $ do
+        parse parsePatternMatches "" "0 -> 0 | 1 -> 1 | otherwise -> 2" `shouldBe` 
+                        Right (FullPatternMatch 
+                                [ PatternMatchLit (PLiteral (IntLiteral 0)) (IntLiteral 0)
+                                , PatternMatchLit (PLiteral (IntLiteral 1)) (IntLiteral 1) ]
+                                (OtherwiseLit (IntLiteral 2)))
+
+    it "parses multiple Integer -> Bool match cases separated by | and otherwise" $ do
+        parse parsePatternMatches "" "0 -> False | 1 -> True | otherwise -> True" `shouldBe` 
+                        Right (FullPatternMatch 
+                                [ PatternMatchLit (PLiteral (IntLiteral 0)) (BoolLiteral False)
+                                , PatternMatchLit (PLiteral (IntLiteral 1)) (BoolLiteral True) ]
+                                (OtherwiseLit (BoolLiteral True)))
+
+    it "parses multiple String -> Bool match cases separated by | and otherwise" $ do
+        parse parsePatternMatches "" "\"Haskell\" -> True | \"Java\" -> True | otherwise -> False" `shouldBe` 
+                        Right (FullPatternMatch 
+                                [ PatternMatchLit (PLiteral (StringLiteral "Haskell")) (BoolLiteral True)
+                                , PatternMatchLit (PLiteral (StringLiteral "Java")) (BoolLiteral True) ]
+                                (OtherwiseLit (BoolLiteral False)))
+
+    it "parses multiple String -> Integer match cases separated by | and otherwise" $ do
+        parse parsePatternMatches "" "\"Zero\" -> 0 | \"One\" -> 1 | otherwise -> 2" `shouldBe` 
+                        Right (FullPatternMatch 
+                                [ PatternMatchLit (PLiteral (StringLiteral "Zero")) (IntLiteral 0)
+                                , PatternMatchLit (PLiteral (StringLiteral "One")) (IntLiteral 1) ]
+                                (OtherwiseLit (IntLiteral 2)))
+
+    it "parses multiple Bool -> String match cases separated by | and otherwise" $ do
+        parse parsePatternMatches "" "True -> \"Hello\" | otherwise -> \"Bye\"" `shouldBe` 
+                        Right (FullPatternMatch 
+                                [ PatternMatchLit (PLiteral (BoolLiteral True)) (StringLiteral "Hello") ]
+                                (OtherwiseLit (StringLiteral "Bye")))
+
+    -- it "parses multiple Double -> String match cases separated by | and otherwise" $ do
```

Comment:
You have a lot of code commented about the Double I think is a mistake Please fix it that

Reply:
Yeah, it was due to the origin branch was not merged yet into develop so the changes were old ones, but now it's okay.